src/main/java/com/samoilenko/onlinebookstore/mapper/BookMapper.java

```diff
@@ -1,18 +1,33 @@
 package com.samoilenko.onlinebookstore.mapper;
 
 import com.samoilenko.onlinebookstore.config.MapperConfig;
-import com.samoilenko.onlinebookstore.dto.BookDto;
-import com.samoilenko.onlinebookstore.dto.BookRequestDto;
+import com.samoilenko.onlinebookstore.dto.bookdtos.BookDto;
+import com.samoilenko.onlinebookstore.dto.bookdtos.BookDtoWithoutCategoryIds;
+import com.samoilenko.onlinebookstore.dto.bookdtos.BookRequestDto;
 import com.samoilenko.onlinebookstore.model.Book;
+import java.util.stream.Collectors;
+import org.mapstruct.AfterMapping;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
 
 @Mapper(config = MapperConfig.class)
 public interface BookMapper {
 
+    @Mapping(target = "categories", ignore = true)
     BookDto toDto(Book book);
 
     @Mapping(target = "id", ignore = true)
     @Mapping(target = "deleted", ignore = true)
-    Book toModel(BookRequestDto requestDto);
+    @Mapping(target = "categories", ignore = true)
+    Book toEntity(BookRequestDto requestDto);
```

Comment:
How do you save the categories you pass into the controller in your database, given that you're not mapping categories here?